dessia_common/exports.py

```diff
@@ -253,3 +256,100 @@ def autosize_sheet_columns(sheet, min_width=5, max_width=30):
             if width > 0:
                 adjusted_width = min((width + 0.5), max_width)
                 sheet.column_dimensions[column].width = adjusted_width
+
+
+class MarkdownWriter:
+    def __init__(self, print_limit: int = 25, table_limit: int = 12):
+        self.print_limit = print_limit
+        self.table_limit = table_limit
+
+    def _object_titles(self):
+        return ['Attribute', 'Type', 'Value'] #, 'Subvalues']
+
+    def _sequence_to_str(self, value: List[Union[list, dict, set]]):
+        in_values = value
+        if len(value) == 0:
+            return f"empty {type(value).__name__}"
+
+        printed_string = f"{len(value)} "
+        if isinstance(value, dict):
+            in_values = list(value.values())
+
+        all_class_names = list(set(subvalue.__class__.__name__ for subvalue in in_values))
+        if len(all_class_names) == 1:
+            printed_string += ''.join([f"{all_class_names[0]}",
+                                       f"{'s' if len(value) > 1 else ''}"])
+        else:
+            str_all_class = str(all_class_names).translate(str(all_class_names).maketrans('', '', "{}'"))
+            printed_string += f"{str_all_class}"
+
+        return printed_string
+
+    def _object_to_str(self, value) -> str:
+        if hasattr(value, 'name' ):
+            return f"{value.name}"
+        return 'unnamed'
+
+    def _value_to_str(self, value: Any) -> str:
+        if isinstance(value, (float, int, bool, complex)):
+            return str(round(value, 6))
+        if isinstance(value, str):
+            return (value if value != '' else 'no_value')
+        if isinstance(value, (list, dict, set)):
+            return self._sequence_to_str(value)
+        return self._object_to_str(value)
+
+    def _string_in_table(self, string: str = ''):
+        return string[:self.print_limit] + ('...' if len(string) > self.print_limit else '')
+
+    def _object_matrix(self, object_):
+        matrix = []
+        for attr, value in object_.__dict__.items():
+            matrix.append([attr,
+                           type(value).__name__,
+                           # (self._sequence_to_str(value) if isinstance(value, (list, dict, set)) else ' - '),
+                           (self._value_to_str(value) if not isinstance(value, (list, dict, set)) else ' - ')])
+        return matrix
+
+    def _head_table(self, col_names: List[str] = None) -> str:
+        cap_names = map(lambda x: x.capitalize(), col_names)
+        return ("| " + " | ".join(cap_names) + " |\n" +
+                "| ------ " * len(col_names) + "|\n")
+
+    def _table_line(self, row: List[Any]) -> str:
+        line = "|"
+        for value in row:
+            line += ''.join([" ", self._string_in_table(self._value_to_str(value)), " |"])
```

Comment:
Why not `" | ".join([self._string_in_table(self._value_to_str(value))])` ?

Reply:
I write 
`line += ''.join([" ", self._string_in_table(self._value_to_str(value)), " |"])`
instead of
`line += " " + self._string_in_table(self._value_to_str(value)) + " |"`
which is not the same as suggested

Reply:
Hmmm, even I am struggling right now to understand my previous comment. I think I mixed two ideas there.
I understand that joining rather than summing string is more efficient, but is this particular feature really that time consuming ? Otherwise, I don't think we should trade readability over performance.
I am suggesting something else, then, which looks way more correct than my first try : 
```
line = "|"
for stuff:
    line = line + f" {self._string_in_table(self._value_to_str(value))} |"
```

OR, if you really don't want to add strings, I suggest you build another list comprehension beforehand : 

```
all_strings = [self._string_in_table(self._value_to_str(v)) for v in row]
line = " | ".join(all_strings)
line = f"| {line} |" # Si toutefois il faut rajouter des | au début et à la fin. On peut faire ces deux dernières lignes en une seule, mais c'est plus dur à lire, je trouve
```

Which I think was my idea in the first place and would give, if I am not mistaken, the following result : 
`| awesome_string0 | awesome_string1 | ... | awesome_stringn |`
is it the intention behind it ?

Reply:
```
line = "|"
for stuff:
    line = line + f" {self._string_in_table(self._value_to_str(value))} |"
```

This one is very better